### ISSUE TITLE
KeePassXC for Windows hangs when saving file on mapped drive

### ISSUE THREAD

**Issue by pkt-nspktr** [keepassxreboot/keepassxc#11588]
### Have you searched for an existing issue?

- [x] Yes, I tried searching and reviewed the pinned issues

### Brief Summary

It is possible to open a kdbx file from a mapped network drive, but trying to save changes results in KeePassXC hanging and needing to be force-closed, resulting in changes not being saved to the kdbx database.

I am not using KeeShare at this time and the file is not being accessed by any other users (this is a home network and my computer is currently the only one using the NAS at this time).

I can use Notepad++ to modify a text file within the same directory on the mapped network drive as "passdb.kdbx" without issue. I can use LibreOffice can create and modify files on the mapped drive without issue. I infer it is not a permissions issue by the fact I can use other applications to modify files on the mapped drive.

https://github.com/keepassxreboot/keepassxc/issues/803 - this appears to be the closest issue matching my own, but it is from 2017, on linux and several versions of KeePassXC ago. I am using a Linux SAMBA server, but a Microsoft Windows client, so the exact scenario is different.

### Steps to Reproduce

1. Map a network drive (SMB) via Windows Explorer 

- 	`\\nas\Home -> H:`

2. Navigate to kdbx file on mapped drive

- 	`H:\path\to\passdb.kdbx`

3. Double-click on kdbx file to open KeePassXC, or right-click, select Open
4. Unlock kdbx file with passphrase
5. Make a change to the database
6. Click on disk icon, or use Ctrl-S to save the database

### Expected Behavior

The changes to the kdbx database file will be saved to the mapped drive and the application will be available to interact with the open database further.

### Actual Behavior

The UI goes non-responsive: all groups and entries are grayed out; title bar is grayed out; clicking in the window changes the title to "Passwords* - KeePassXC (Not Responding)"

The application can be closed out by clicking on the red X on window, resulting in the changes NOT being saved to the database.

### KeePassXC Debug Information

```Text
KeePassXC - Version 2.7.9
Revision: 8f6dd13

Qt 5.15.11
Debugging mode is disabled.

Operating system: Windows 10 Version 2009
CPU architecture: x86_64
Kernel: winnt 10.0.19045

Enabled extensions:
- Auto-Type
- Browser Integration
- Passkeys
- SSH Agent
- KeeShare
- YubiKey
- Quick Unlock

Cryptographic libraries:
- Botan 3.1.1

Current KeePassXC Settings:
General > Basic Settings > File Management
	[ ] Automatically save after every change
	[X] Automatically save when locking database
	[X] Automatically save non-data changes when locking database
	[X] Automatically reload the database when modified externally
	[X] Use alternative saving method (may solve problems with Dropbox, Google Drive, GVFS, etc.)
		[[ Temporary file moved into place ]]

The above behavior results if the "Use alternative saving method" setting is changed to [[ Directly write to database file (dangerous) ]] or disabled altogether
```

### Operating System

Windows

### Linux Desktop Environment

None

### Linux Windowing System

None

**Comment by droidmonkey** [comment]
I just did this and it worked fine☹

Home NAS, mounted to Z:/, opened and saved without any issues. For what it is worth, saving **should** timeout if it is a network issue. Not sure how long you waited for it to finish.

**Comment by pkt-nspktr** [comment]
Thank you for confirming that this use case should be working.

I see there is a "--debug-info" command line argument, but I've not been able to see where the debug information is displayed. Despite running with the "debug-info" option, the "debugging mode" is reported as being disabled.

Based on the feedback about timeout, I decided to wait to see what happened; I was normally waiting ~30 seconds before killing KeePassXC. I let it run for over 5 minutes and it maintained a non-responsive state. 

I looked at the SMB config options provided by the NAS and enabled debug logging. Once that logging was enabled, KeePassXC became responsive, enabling modifying and saving the password database without a noticeable delay. I disabled logging, and then KeePassXC returned to the reported behavior (staying non-responsive attempting to saving a change to the kdbx file).

I then enabled the option to "Disable Opportunistic locking". Applying this setting allowed KeePassXC to modify the kdbx files without delay, just like enabling logging had. 

So this appears to have been a SMB issue rather than a KeePassXC issue, but it would have been convenient to have KeePassXC provide some debugging information about file locking that may have helped identify the source of this issue sooner.

Hopefully this information will help others who may run into this issue.
